packages/zoe/src/contractFacet/vatRoot.js

```diff
@@ -8,51 +8,84 @@
 // `yarn build-zcfBundle`.
 
 import { Far } from '@endo/marshal';
+import { E } from '@endo/far';
 
 import '../../exported.js';
 import '../internal-types.js';
 
 import { makeZCFZygote } from './zcfZygote.js';
 
+const { details: X } = assert;
+
 /**
  * @param {VatPowers & { testJigSetter: TestJigSetter }} powers
- * @returns {{ executeContract: ExecuteContract}}
+ * @param {{contractBundleCap: BundleCap, zoeService: ZoeService, invitationIssuer: Issuer, privateArgs: any}} vatParameters
+ * @param {import('@agoric/vat-data').Baggage} baggage
  */
-export function buildRootObject(powers) {
+// * @returns {{ executeContract: ExecuteContract}}
+export async function buildRootObject(powers, vatParameters, baggage) {
   // Currently, there is only one function, `executeContract` called
   // by the Zoe Service. However, when there is kernel support for
   // zygote vats (essentially freezing and then creating copies of
   // vats), `makeZCFZygote`, `zcfZygote.evaluateContract` and
   // `zcfZygote.startContract` should exposed separately.
   const { testJigSetter } = powers;
+  const { contractBundleCap } = vatParameters;
+  assert(
+    contractBundleCap !== undefined,
```

Comment:
is there any falsy value that's legit?
```suggestion
    contractBundleCap
```

Reply:
done